.github/workflows/sketch-compile.yml

```diff
@@ -0,0 +1,53 @@
+name: Sketch Compile
+
+# the action is triggered at every pull request
+on:
+  - pull_request
+
+jobs:
+  sketch-compile:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        fqbn:
+          - arduino:samd:mkrwifi1010
+          - arduino:samd:nano_33_iot
+          - arduino:megaavr:uno2018:mode=on
+          - arduino:mbed:nano33ble
+          - arduino:mbed_nano:nanorp2040connect
+          - arduino:renesas_uno:unor4wifi
+          - "esp8266:esp822:d1_mini"
+
+    steps:
+      # This step makes the contents of the repository available to the workflow
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # For more information: https://github.com/arduino/compile-sketches#readme
+      - name: Compile sketch
+        uses: arduino/compile-sketches@v1
+        with:
+          # The default is to compile for the Arduino Uno board. If you want to compile for other boards, use the `fqbn` input.
+          sketch-paths: |
+            # search the sketches in the current repo (no subfolders)
+            - ./
+
+          platforms: |
+            - name: "arduino:samd"
+            - name: "arduino:megaavr"
+            - name: "arduino:mbed"
+            - name: "arduino:mbed_nano"
+            - name: "arduino:renesas_uno"
```

Comment:
@Frack91 here also. I think the implicit usage of WiFi could not be fully compatible with an arduino uno or even worse, with a nano